packages/app/src/specs/SpecsList.vue

```diff
@@ -417,6 +420,20 @@ const { refetchFailedCloudData } = useCloudSpecData(
   props.gql.currentProject?.specs as SpecsListFragment[] || [],
 )
 
+const { runAllSpecs, isRunAllSpecsAllowed } = useRunAllSpecs()
```

Comment:
This will launch an additional GraphQL query - was this the intention? Should we follow the pattern of using a `fragment` and spreading it over the top level query, so we just have a single query for the whole page?

I'm not sure we have finalized the pattern here, but this is what we are doing so far in most of the application.

I'm guessing you set this up with the intention of making it re-usable in the runner page, too. Can you speak a bit more to how you see this working across both the spec page and the runner page?

Reply:
It doesn't launch an additional graphql query, since both the `config` and `currentTestingType` is already in the cache so it can read it from there (you can also check the network logs). I've seen a shift away from the previous pattern e.g. [CloudViewerAndProject](https://github.com/cypress-io/cypress/blob/21315b0044069f837399b27658737cc368d0c960/packages/frontend-shared/src/gql-components/CloudViewerAndProject.vue#L84) and I personally like having a composable that fully encapsulates all the logic necessary for a certain task.

Even if it did make another network request, with how graphql is set up the cost would be very minimal (websocket request, only resolve a minimal amount of fields). I can refactor to use the previous pattern if you think this pattern doesn't scale, but the fragment spreading is also really annoying :(

I had this plugged up to both spec lists previously, almost exactly the same way. I didn't want to fix the UI up so I pulled it from there. The InlineSpecsList can just import this composable and do similar logic to filter the specs. The spec filtering could be pulled into the composable but I knew the UI work was looking to change how the tree was being built so I didn't want to pull too much in.

Reply:
Ah yeah, I figured you were relying on the cache to be populated. I think this is probably fine then - I agree spreading all the fragments is pretty annoying. I wonder if we can have a watcher automatically spread the fragments, that'd be a nice DX.

Anyway, let's move forward with this for now. I think this pattern is actually fine. I hope we can settle on "one way to do things" as we move forward, we really need to get a handle on App, it's getting messy.